root_VS2012/programs/C#/Samples/WebApp_sample/MVC_Sample/MVC_Sample/Framework/Js/common.js

```diff
@@ -0,0 +1,759 @@
+// Ajax Extensionの開始前イベント処理
+// ---------------------------------------------------------------
+// 引数    sender, args
+// ---------------------------------------------------------------
+function Fx_AjaxExtensionInitializeRequest(sender, args) {
+
+    // これが呼ばれるのは、Fx_OnSubmitの後
+
+    // ∴ 下記の処理は、Fx_OnSubmitの二重送信防止機能に統合
+
+    // // 現在、実行中の非同期通信が存在するかを判定
+    // if (pageRequestManager.get_isInAsyncPostBack())
+    // { 
+    //     // 非同期通信中である場合にはエラー メッセージを表示
+    //     alert("二重送信です(ajax)");
+    // 
+    //     // 後続の処理をキャンセル
+    //     args.set_cancel(true);
+    // }
+
+    // ★★ Fx_OnSubmitが呼ばれるのは、Ajax Extensionのみ。
+    // 　　 ClientCallbackや、WebServiceBridgeでは、呼ばれない。
+
+    // ポストバック エレメントを取得
+    AjaxPostBackElement = args.get_postBackElement();
+
+    if (AjaxPostBackElement) {
+        AjaxPostBackElement.disabled = true;
+    }
+
+    // 二重送信フラグの設定
+    Ajax_IsProgressed = true;
+    // ★★ ここの処理が動く前に、
+    // 他が、Fx_OnSubmitを通過しないか？
+}
+
+// ---------------------------------------------------------------
+// Ajax Extensionの終了後イベント処理
+// ---------------------------------------------------------------
+// 引数    sender, args
+// ---------------------------------------------------------------
+function Fx_AjaxExtensionEndRequest(sender, args) {
+    // はじめにタイマをクリアする。
+    clearTimeout(ProgressDialogTimer);
+
+    // プログレス ダイアログを非表示にする。
+    try {
+
+        document.body.removeChild(AjaxMask);
+        document.body.removeChild(AjaxProgressDialog);
+
+    } catch (e) {
+        //alert( e );//エラー内容
+    }
+
+    // 二重送信フラグの設定
+    Ajax_IsProgressed = false;
+
+    // イベント発生元の要素を有効化
+
+    if (AjaxPostBackElement) {
+        AjaxPostBackElement.disabled = false;
+    }
+}
+
+// ---------------------------------------------------------------
+// ClientCallbackや、WebServiceBridge、
+// XMLHTTPRequest、jQueryなど、Ajax非同期通信用
+// 汎用の開始前処理、終了後処理関数を用意する（予定）。
+// 
+// 技術毎に、開始終了処理のイベント実装の機構を持つが、
+// それを使うか、どうか、などの検討も必要になる。
+// ---------------------------------------------------------------
+
+//**********************************************************************************
+
+// ---------------------------------------------------------------
+//  ユーティリティ：element取得関数
+// ---------------------------------------------------------------
+
+// ---------------------------------------------------------------
+// elementをnameの後方一致で取得関数する関数
+// ---------------------------------------------------------------
+// 引数    elementのname（後方一致）
+// 戻り値  取得したelement
+// ---------------------------------------------------------------
+function GetElementByName_SuffixSearch(name) {
+
+    var elementName = "";
+
+    var nameLength = name.length;
+    var elementNameLength = 0;
+
+    var fobj = document.aspnetForm;
+
+    if (fobj == null || fobj == undefined) {
+        fobj = document.getElementById("form1");
+    }
+
+    if (fobj == null || fobj == undefined) {
+        fobj = document.getElementById("aspnetForm");
+    }
```

Comment:
In case of ASP.NET MVC, the number of forms included in the view is not fixed.
Therefore, "fobj" may remain null or undefined.